operator/crdgen/schemagen.go

```diff
@@ -167,6 +167,19 @@ func (generator *SchemaGenerator) traverseInner(message *Message) (*Schema, erro
 			prop.Nullable = true
 		}
 
+		// Labels are relying on `utils.Strings`, which can either marshall as an array of strings or a single string
+		// This does not pass Schema validation from the apiserver, to workaround we don't specify type for those fields
+		// and ask Kubernetes to preserve unknown fields.
+		if field.CustomType() == "Labels" {
```

Comment:
The issue only happens when we use custom types that don't marshall consistently. As we're inconsistant in our inconsistencies `User.Traits` are not leveraging the `utils.Strings` type, hence everything is ok.

I added similar YAMLSpec tests for users anyway as this will be handy if we need to reproduce issues.